export const GlobalTexts = {
    selectPlaceholder: "Select...",
}

export const GenomeOverviewTexts = {
    title: "Download genomes",
    titleComment: "Here you can download genomes based on filters",
    projectSelector: {
        selectProjectLabel: "Select project",
        selectSampleLabel: "Select sample",
        allProjects: "All projects",
        allSamples: "All samples",
    },
    downloadLinks: {
        downloadFasta: "Download fasta file",
        downloadMetadata: "Download metadata csv"
    },
    filters: {
        title: "Filters",
        genomeQualityLabel: "Genome quality",
        genomeQualityLow: "Only Low",
        genomeQualityMedium: "Only Medium",
        genomeQualityHigh: "Only High",
        passGnucLabel: "Pass GNUC",
        passGnucAll: "All GNUC",
        passGnucYes: "Yes",
        passGnucNo: "No",
        clearFiltersButton: "Clear all filters",
        completenessLabel: "Completeness",
        contaminationLabel: "Contamination",
        trnaLabel: "tRNA",
        s16Label: "16S",
        s5Label: "5S",
        s23Label: "23S",
        highQualityGenomesCountLabel: "High quality genomes: ",
        mediumQualityGenomesCountLabel: "Medium quality genomes: ",
        lowQualityGenomesCountLabel: "Low quality genomes: ",
    },
    pagination: {
        itemsPerPage: "Items per page",
        itemsCountLabel: "%{from}-%{to} of %{count} items",
        pagesCountLabel: "of %{count}",
        previous: "Previous",
        next: "Next",
    },
    table: {}
}

export const genomeDetailsTexts = {
    genome: "Genome",
    download: "Download",
    genomeSize: "Genome size",
    completeness: "Completeness",
    contamination: "Contamination",
    gcContent: "GC content",
    totalCodingSequences: "Total coding sequences",
    numberOfGenes: "Number of genes",
    called: "Called",
    mapped: "Mapped",
    contigComposition: "Contig composition",
    nContigs: "Number of contigs",
    contigN50: "Contig N50",
    additionalNotes: "Additional Notes",
    classification: "Classification",
    msaPercent: "MSA percent",
    redValue: "RED value",
    proportionGenesRetainedInMajorClades: "Proportion of genes retained in major clades",
    genesRetainedIndex: "Genes retained index",
    cladeSeparationScore: "Clade separation score",
    contaminationPortion: "Contamination portion",
    nEffectiveSurplusClades: "Number of effective surplus clades",
    meanHitIdentity: "Mean hit identity",
    referenceRepresentationScore: "Reference representation score",
    passGNUC: "Pass.GNUC",
    trnaNumber: "Number of tRNA",
    s16Number: "Number of 16S",
    s5Number: "Number of 5S",
    s23Number: "Number of 23S",
    expand: "Expand",
    classificationLabel: "Classification",
    showTableLabel: "Show table",
    hideTableLabel: "Hide table",
    domain: "Domain",
    phylum: "Phylum",
    class: "Class",
    order: "Order",
    family: "Family",
    genus: "Genus",
    species: "Species",
    unknown: "Unknown",
}
